### PR TITLE
38 clippy

### PR DESCRIPTION
**Pull request by sgreenbury**
Add clippy checks to pre-commit and actions (#38).

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -10,3 +10,4 @@ repos:
     hooks:
     -   id: fmt
     -   id: cargo-check
+    -   id: clippy
```
Comment (sgreenbury) [inline .pre-commit-config.yaml:13]: To discuss whether this hook is helpful or will be inhibitory.

File: trustchain-core/src/commitment.rs [hunk 2]
```diff
@@ -11,6 +11,12 @@ use thiserror::Error;
 /// Type for commitment result.
 pub type CommitmentResult<T> = Result<T, CommitmentError>;
 
+/// Type for commitment function.
+pub type CommitmentFunction = dyn Fn(&serde_json::Value) -> CommitmentResult<Value>;
```
[the remark this answers is not in the document]
Comment (sgreenbury) [reply trustchain-core/src/commitment.rs:15]: We could alternatively specify our prefered clippy params in a `.clippy.toml`

[the remark this answers is not in the document]
Comment (sgreenbury) [reply trustchain-core/src/commitment.rs]: Resolved with `.clippy.toml`

File: .pre-commit-config.yaml (hunk 1 above)
Comment (sgreenbury) [reply .pre-commit-config.yaml]: Removed

**Review by thobson88 — APPROVED** [review]
LATM

**Pull request merged**